### ISSUE TITLE
Error when grad is None but param is not None

### ISSUE THREAD

**Issue by varun-parthasarathy** [kozistr/AdaBound-tensorflow#6]
When I use the optimizer, there are some cases where the gradient is None while the parameter isn't. Due to this, I get a ValueError, as the None type object cannot be converted to a Tensor (this arises in the calculation of m_t and v_t). I think the 'and' condition on line 76 should be made an 'or' condition instead.

Also, I had a question regarding the weight decay part. It looks like the weight decay is being added to the weights instead of subtracted on line 124. Could you please clarify this?

**Comment by kozistr** [comment]
1. oh, yeap, you are right. i will fix it for that case!

2. as i know, the term 'weight decay' (a.k.a weight regularization) usually defined as a term added directly to the update rule. so, 'add' operation seems to be better than 'subtract' operation.

if you have more questions, feel free to re/open an issue and comment : )

thanks in advance!

**Comment by varun-parthasarathy** [comment]
Now that you mention it, I think you're right. I was looking at it more along the lines of decoupled weight decay instead. Thanks for your clarifications!
